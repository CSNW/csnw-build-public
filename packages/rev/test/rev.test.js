const { join } = require('path');
const { Files } = require('@csnw-build/pipeline');

const rev = require('../')();
const cwd = join(__dirname, 'fixtures');

test('skip pipeline in development', async () => {
  const process = await rev.pipeline({
    mode: 'development',
    cwd,
    sourcemaps: true,
    cache: false
  });
  const files = await Files.from('*', { cwd });

  const result = await process(files);
  expect(result).toMatchSnapshot();
});

test('pipeline', async () => {
  const process = await rev.pipeline({
    mode: 'production',
    cwd,
    cache: false,
    sourcemaps: true
  });
  const files = await Files.from('*', { cwd });

  const result = await process(files);
  expect(result).toMatchSnapshot();
});
