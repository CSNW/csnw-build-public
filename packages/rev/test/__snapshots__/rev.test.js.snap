// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pipeline 1`] = `
Object {
  "a.html": Object {
    "baseDir": "C:/dev/csnw-build/packages/rev/test/fixtures",
    "contents": "<!doctype html>
<html>
  <head>
    <link href=\\"./c-8171d94d61.css\\" rel=\\"stylesheet\\">
  </head>
  <body>
    <script src=\\"b-bb454f7a0d.js\\"></script>
  </body>
</html>",
    "cwd": "C:/dev/csnw-build/packages/rev/test/fixtures",
    "path": "C:/dev/csnw-build/packages/rev/test/fixtures/a.html",
    "sourceMap": null,
  },
  "b-bb454f7a0d.js": Object {
    "baseDir": "C:/dev/csnw-build/packages/rev/test/fixtures",
    "contents": "var name = 'b';
",
    "cwd": "C:/dev/csnw-build/packages/rev/test/fixtures",
    "path": "C:/dev/csnw-build/packages/rev/test/fixtures/b-bb454f7a0d.js",
    "sourceMap": Object {
      "file": "b.js",
      "mappings": "AAAA;",
      "names": Array [],
      "sources": Array [
        "b.js",
      ],
      "sourcesContent": Array [
        "var name = 'b';
",
      ],
      "version": 3,
    },
  },
  "c-8171d94d61.css": Object {
    "baseDir": "C:/dev/csnw-build/packages/rev/test/fixtures",
    "contents": ".logo {
  background: url(d-ad30464376.svg);
}
",
    "cwd": "C:/dev/csnw-build/packages/rev/test/fixtures",
    "path": "C:/dev/csnw-build/packages/rev/test/fixtures/c-8171d94d61.css",
    "sourceMap": Object {
      "file": "c.css",
      "mappings": "AAAA;kBACkB,gBAAK;;",
      "names": Array [],
      "sources": Array [
        "c.css",
      ],
      "sourcesContent": Array [
        ".logo {
  background: url(d.svg);
}
",
      ],
      "version": 3,
    },
  },
  "d-ad30464376.svg": Object {
    "baseDir": "C:/dev/csnw-build/packages/rev/test/fixtures",
    "contents": "<text>Howdy</text>",
    "cwd": "C:/dev/csnw-build/packages/rev/test/fixtures",
    "path": "C:/dev/csnw-build/packages/rev/test/fixtures/d-ad30464376.svg",
    "sourceMap": null,
  },
  "manifest.json": Object {
    "baseDir": "C:/dev/csnw-build/packages/rev/test/fixtures",
    "contents": "{
  \\"b.js\\": \\"b-bb454f7a0d.js\\",
  \\"d.svg\\": \\"d-ad30464376.svg\\",
  \\"c.css\\": \\"c-8171d94d61.css\\"
}",
    "cwd": "C:/dev/csnw-build/packages/rev/test/fixtures",
    "path": "C:/dev/csnw-build/packages/rev/test/fixtures/manifest.json",
    "sourceMap": null,
  },
}
`;

exports[`skip pipeline in development 1`] = `
Object {
  "a.html": Object {
    "baseDir": "C:/dev/csnw-build/packages/rev/test/fixtures",
    "contents": "<!doctype html>
<html>
  <head>
    <link href=\\"./c.css\\" rel=\\"stylesheet\\">
  </head>
  <body>
    <script src=\\"b.js\\"></script>
  </body>
</html>",
    "cwd": "C:/dev/csnw-build/packages/rev/test/fixtures",
    "path": "C:/dev/csnw-build/packages/rev/test/fixtures/a.html",
    "sourceMap": null,
  },
  "b.js": Object {
    "baseDir": "C:/dev/csnw-build/packages/rev/test/fixtures",
    "contents": "var name = 'b';
",
    "cwd": "C:/dev/csnw-build/packages/rev/test/fixtures",
    "path": "C:/dev/csnw-build/packages/rev/test/fixtures/b.js",
    "sourceMap": null,
  },
  "c.css": Object {
    "baseDir": "C:/dev/csnw-build/packages/rev/test/fixtures",
    "contents": ".logo {
  background: url(d.svg);
}
",
    "cwd": "C:/dev/csnw-build/packages/rev/test/fixtures",
    "path": "C:/dev/csnw-build/packages/rev/test/fixtures/c.css",
    "sourceMap": null,
  },
  "d.svg": Object {
    "baseDir": "C:/dev/csnw-build/packages/rev/test/fixtures",
    "contents": "<text>Howdy</text>",
    "cwd": "C:/dev/csnw-build/packages/rev/test/fixtures",
    "path": "C:/dev/csnw-build/packages/rev/test/fixtures/d.svg",
    "sourceMap": null,
  },
}
`;
